### PR TITLE
Add authenticate with TOTP method

### PR DESCRIPTION
**Pull request by alisherry**
## Description
Add functionality to authenticate a user using `TOTP` 

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[X] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/workos/user_management.rb [hunk 1]
```diff
@@ -462,6 +462,40 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
         )
         WorkOS::UserResponse.new(response.body)
       end
+      #
```
Comment (mthadley) [inline lib/workos/user_management.rb:465]: Can we leave an extra blank line above this comment block like we've done for other methods?

```suggestion

```

File: lib/workos/user_management.rb [hunk 2]
```diff
@@ -462,6 +462,40 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
...
+      # @param [String] user_agent The user agent of the request from the user who is attempting to authenticate.
+      #
+      # @return WorkOS::UserResponse
+
+      sig do
+        params(
+          code: String,
+          client_id: String,
+          pending_authentication_token: String,
+          authentication_challenge_id: String,
```
Comment (mthadley) [inline lib/workos/user_management.rb:480]: Looks like we are missing `ip_address` and `user_agent` as optional parameters:

```suggestion
          code: String,
          client_id: String,
          pending_authentication_token: String,
          authentication_challenge_id: String,
          ip_address: T.nilable(String),
          user_agent: T.nilable(String),
```

File: lib/workos/user_management.rb [hunk 3]
```diff
@@ -462,6 +462,40 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
...
         WorkOS::UserResponse.new(response.body)
       end
+      #
+      # Authenticates a user using TOTP.
+      #
+      # @param [String] code The one-time code that was emailed to the user.
+      # @param [String] client_id The WorkOS client ID for the environment
+      # @param [String] ip_address The IP address of the request from the user who is attempting to authenticate.
+      # @param [String] user_agent The user agent of the request from the user who is attempting to authenticate.
+      #
```
Comment (mthadley) [inline lib/workos/user_management.rb:472]: We are missing a `@param` for `pending_authentication_token` and `authentication_challenge_id`.

File: lib/workos/user_management.rb (hunk 2 above)
Comment (alisherry) [reply lib/workos/user_management.rb:480]: Derp, actually this is just bad doc copy pasted on my part. Will remove these from the description docs

**Review by mthadley — APPROVED** [review]

**Pull request merged**